eclipse-collections-code-generator/src/main/resources/impl/map/mutable/primitivePrimitiveHashMap.stg

```diff
@@ -477,6 +477,54 @@ public class <name1><name2>HashMap extends AbstractMutable<name2>ValuesMap imple
         <getIfAbsentPut("")>
     }
 
+    @Override
+    public <type2> getAndPut(<type1> key, <type2> putValue, <type2> defaultValue)
+    {
+        if (isEmptyKey(key))
+        {
+            if (this.sentinelValues == null)
```

Comment:
I think this could be simplified as follows: 

```
            if (this.sentinelValues == null)
            {
                this.sentinelValues = new SentinelValues();
            }
            else if (this.sentinelValues.containsZeroKey)
            {
                <type2> existingValue = this.sentinelValues.zeroValue;
                this.sentinelValues.zeroValue = putValue;
                return existingValue;
            }
            this.addEmptyKeyValue(putValue);
            return defaultValue;
```

Reply:
Simplified. Thanks.